parsec/parsec_internal.h

```diff
@@ -590,7 +590,7 @@ extern int device_delegate_begin, device_delegate_end;
                            key_hash((TASK)->task_class->make_key(           \
                               (TASK)->taskpool, (TASK)->locals ), NULL),    \
                            (TASK)->taskpool->taskpool_id,                   \
-                           (TASK)->task_class->profile_info,                \
+                           (TRACE_INFO) ? (TASK)->task_class->profile_info : NULL, \
```

Comment:
why is the `task_class->profile_info` not initialized to NULL ?

Reply:
profile_info points to the function defined by the task class.

TRACE_INFO is a macro argument that decides if we should log the info now (typically by PINS, either at the begin or the end of the event).

With this change, we keep the previous functionality (if the task class defines a profile_info function, PINS decides if we log the info or not), and we support the case where the task class has not defined a profile_info function (DTD and PTG always define one, because there is always at least the basic info of the task, but MADNESS over parsec doesn't define one).